### PR TITLE
[RTC-456] Add peerless room purge

### PR DESCRIPTION
**Pull request by sgfn**
## Acknowledging the stipulations set forth:
 - [x] I hereby confirm that a Pull Request involving updates to the Software Development Kit (SDK) has been smoothly merged, currently awaits processing, or is otherwise deemed unnecessary in this context.
 - [x] I also affirm that another Pull Request, specifically addressing updates to the documentation body (commonly referred to as 'docs'), has either been successfully incorporated, is in the process of review, or is considered superfluous under the prevailing circumstances.

### CODE REVIEW

File: test/jellyfish_web/controllers/room_controller_test.exs [hunk 1]
```diff
@@ -118,22 +118,6 @@ defmodule JellyfishWeb.RoomControllerTest do
...
-      response = json_response(conn, :ok)
-      assert_response_schema(response, "RoomDetailsResponse", @schema)
-
-      assert %{
-               "id" => ^id,
-               "config" => %{},
-               "components" => [],
-               "peers" => []
-             } = response["data"]
-    end
```
Comment (Rados13) [inline test/jellyfish_web/controllers/room_controller_test.exs:135]: Why did you remove this test?

**Review by Rados13 — APPROVED** [review]

File: test/jellyfish_web/controllers/room_controller_test.exs (hunk 1 above)
Comment (sgfn) [reply test/jellyfish_web/controllers/room_controller_test.exs:135]: It doesn't add anything of value, the checks performed here are already done in "renders room when data is valid" and "renders room when data is valid, custom room_id"

File: test/jellyfish/room_test.exs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+defmodule Jellyfish.RoomTest do
+  use ExUnit.Case, async: true
+
+  alias Jellyfish.{Peer, Room}
+
+  @purge_timeout_s 1
```
Comment (mickel8) [inline test/jellyfish/room_test.exs:6]: The way those tests are written right now will cause us to wait a couple of seconds until they are executed. I am wondering if we could make this better. We can discuss this offline

File: lib/jellyfish/room.ex [hunk 3]
```diff
@@ -554,6 +560,19 @@ defmodule Jellyfish.Room do
...
   end
 
+  @impl true
+  def handle_info(:peerless_purge, state) do
+    if peerless_long_enough?(state) do
+      Logger.info(
+        "Removing room because it was peerless for #{state.config.peerless_purge_timeout} seconds"
+      )
+
+      {:stop, :normal, state}
```
[the remark this answers is not in the document]
Comment (sgfn) [reply lib/jellyfish/room.ex:570]: But that's exactly how `RoomService` removes rooms -- [here](https://github.com/jellyfish-dev/jellyfish/blob/a403ecc47e50da2163fcdf08f8e333ff607ec4d4/lib/jellyfish/room_service.ex#L263)

We're manually removing all of the linked processes in the `terminate` callback, no?

File: test/jellyfish_web/controllers/room_controller_test.exs (hunk 1 above)
Comment (Rados13) [reply test/jellyfish_web/controllers/room_controller_test.exs:135]: Practically speaking, yes; philosophically, no. It is not the responsibility of these tests to verify that action X occurs, especially if it's not mentioned in their names. However, it is easy to accidentally refactor these tests and unintentionally remove coverage for this edge case. If you insist on removing this test, I suggest adding a note to one of the other tests indicating that it also covers the situation previously handled by the removed test.

Comment (Rados13) [reply test/jellyfish_web/controllers/room_controller_test.exs:135]: Practically speaking, yes; philosophically, no. It is not the responsibility of these tests to verify that action X occurs, especially if it's not mentioned in their names. However, it is easy to accidentally refactor these tests and unintentionally remove coverage for this edge case. If you insist on removing this test, I suggest adding a note to one of the other tests indicating that it also covers the situation previously handled by the removed test or modifying their names.

**Review by Rados13 — APPROVED** [review]

File: test/jellyfish_web/controllers/room_controller_test.exs (hunk 1 above)
Comment (sgfn) [reply test/jellyfish_web/controllers/room_controller_test.exs:135]: Updated name of test

**Comment by sgfn** [comment]
Bumped Elixir to 1.15 because we use Map.intersect which was added in that version

**Review by mickel8 — APPROVED** [review]
